### PR TITLE
[throwtheswitch-unity] Create a new port

### PR DESCRIPTION
**Pull request by nfarid**
**Describe the pull request**

- #### What does your PR fix?
  Adds the port for [Unity (C unit testing library).](https://github.com/ThrowTheSwitch/Unity)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All, I presume.
  No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes.

### CODE REVIEW

File: ports/unity/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,36 @@
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO "ThrowTheSwitch/Unity"
+    REF "v2.5.2"
```
Comment (Adela0814) [inline ports/unity/portfile.cmake:5]:
```suggestion
    REF "v${VERSION}"
```

File: ports/unity/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (Adela0814) [inline ports/unity/portfile.cmake:1]: Please remove this new line.

File: ports/unity/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    FEATURES
+        fixture     UNITY_EXTENSION_FIXTURE
+        memory      UNITY_EXTENSION_MEMORY
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS ${FEATURE_OPTIONS}
+        -DUNITY_EXTENSION_FIXTURE="${UNITY_EXTENSION_FIXTURE}"
+        -DUNITY_EXTENSION_MEMORY="${UNITY_EXTENSION_MEMORY}"
```
Comment (Adela0814) [inline ports/unity/portfile.cmake:22]:
```suggestion
        "-DUNITY_EXTENSION_FIXTURE=${UNITY_EXTENSION_FIXTURE}"
        "-DUNITY_EXTENSION_MEMORY=${UNITY_EXTENSION_MEMORY}"
```

File: ports/unity/usage [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+The package unity provides CMake targets:
+
+    find_package("unity" REQUIRED)
+    target_link_libraries("main" PRIVATE "unity::framework")
```
Comment (Adela0814) [inline ports/unity/usage:5]: The quotation marks here are unnecessary.

File: ports/unity/usage [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (Adela0814) [inline ports/unity/usage:1]: Please remove this line.

File: ports/unity/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "unity",
```
Comment (Adela0814) [inline ports/unity/vcpkg.json:2]: Please add a prefix for the port name.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by nfarid** [comment]
Did the following changes:
- Changed REF to use ${VERSION}
- Moved quotation marks in vcpkg_cmake_configure options
- Remove redundant quotation marks from the usage file
- Changed port's name

**Comment by Adela0814** [comment]
Usage test failed in my side:
failures:
image

File: ports/throwtheswitch-unity/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (Adela0814) [inline ports/throwtheswitch-unity/portfile.cmake:1]: Please remove this blank line.

File: ports/throwtheswitch-unity/usage [hunk 8]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (Adela0814) [inline ports/throwtheswitch-unity/usage:1]: Please remove this blank line.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**